### PR TITLE
Minor: Upate the `sqllogictest` crate README

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Related to https://github.com/apache/arrow-datafusion/issues/7966
Related to https://github.com/apache/arrow-datafusion/pull/7964

## Rationale for this change

While working on https://github.com/apache/arrow-datafusion/pull/7970 I noticed that the sqllogictest crate's README was slightly differently formatted than the others. It would be nice to be consistent

## What changes are included in this PR?

1. Update the introduction of the README to follow the same pattern as the other crate readmes
2. Use standard headings

### CODE REVIEW

File: datafusion/sqllogictest/README.md [hunk 1]
```diff
@@ -17,19 +17,26 @@
   under the License.
 -->
 
-#### Overview
+# DataFusion sqllogictest
 
-This is the Datafusion implementation of [sqllogictest](https://www.sqlite.org/sqllogictest/doc/trunk/about.wiki). We
-use [sqllogictest-rs](https://github.com/risinglightdb/sqllogictest-rs) as a parser/runner of `.slt` files
-in [`test_files`](test_files).
+[DataFusion](df) is an extensible query execution framework, written in Rust, that uses Apache Arrow as its in-memory format.
```
Comment (jonahgao) [inline datafusion/sqllogictest/README.md:22]:
```suggestion
[DataFusion][df] is an extensible query execution framework, written in Rust, that uses Apache Arrow as its in-memory format.
```
Fix the broken link.

**Review by jonahgao — APPROVED** [review]
LGTM!

File: datafusion/sqllogictest/README.md [hunk 2]
```diff
@@ -115,7 +122,7 @@ sqllogictests also supports `cargo test` style substring matches on file names t
 cargo test --test sqllogictests -- information
 ```
 
-#### Running tests: Postgres compatibility
+## Running tests: Postgres compatibility
 
 Test files that start with prefix `pg_compat_` verify compatibility
 with Postgres by running the same script files both with DataFusion and with Posgres
```
Comment (jackwener) [inline datafusion/sqllogictest/README.md:128]:
```suggestion
with Postgres by running the same script files both with DataFusion and with Postgres
```

File: datafusion/sqllogictest/README.md [hunk 3]
```diff
@@ -95,9 +102,9 @@ SELECT * from foo;
 
 Assuming it looks good, check it in!
 
-#### Reference
+# Reference
 
-#### Running tests: Validation Mode
+## Running tests: Validation Mode
 
 In this model, `sqllogictests` runs the statements and queries in a `.slt` file, comparing the expected output in the
```
Comment (jackwener) [inline datafusion/sqllogictest/README.md:109]:
```suggestion
In this mode, `sqllogictests` runs the statements and queries in a `.slt` file, comparing the expected output in the
```

**Review by jackwener — APPROVED** [review]
Thanks @alamb

File: datafusion/sqllogictest/README.md (hunk 1 above)
Comment (jonahgao) [reply datafusion/sqllogictest/README.md:22]: PR to fix broken links in other crates: #7986

**Comment by alamb** [comment]
Thank you @jonahgao  and @jackwener  for the improvements.

**Pull request merged**